### PR TITLE
test(mersenne): add tests for value ranges

### PR DESCRIPTION
**Pull request by ST-DDT**
This PR adds tests asserting that our `MersenneTwister19937` implementation can generate all expected values.

Partially covers #2361 with tests

- #2361

### CODE REVIEW

File: test/faker.spec.ts [hunk 1]
```diff
@@ -113,14 +113,14 @@ describe('faker', () => {
       faker.seed(1);
 
       const actual = faker.animal.cat();
-      expect(actual).toBe('Korat');
+      expect(actual).toMatchInlineSnapshot('"Korat"');
```
Comment (ST-DDT) [inline test/faker.spec.ts]: The inline snapshots are "required" because the value is mersenne and seed specific and the inline snapshots are easier to update.

File: test/mersenne.spec.ts
Comment (ST-DDT) [inline test/mersenne.spec.ts:1]: Renamed to `test/internal/mersenne.spec.ts` and extended with Twister specific tests.

File: test/internal/mersenne.spec.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      const actual = twister.genrandReal2();
+      expect(actual).toBe(0);
+    });
+
+    it('should be able to return almost 1', () => {
+      const twister = newTwister();
+      twister.genrandInt32 = () => 2 ** 32 - 1;
+      const actual = twister.genrandReal2();
+      expect(actual).toBe(TWISTER_32CO_MAX_VALUE);
+    });
```
Comment (xDivisionByZerox) [inline test/internal/mersenne.spec.ts]: I don't understand what these tests are supposed to do.

File: test/internal/mersenne.spec.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    beforeAll(() => {
+      randomizer.seed(seed);
+    });
+
+    describe('next', () => {
+      it('should return random number from interval [0, 1)', () => {
+        const actual = randomizer.next();
+
+        expect(actual).toBeGreaterThanOrEqual(0);
+        expect(actual).toBeLessThanOrEqual(MERSENNE_MAX_VALUE);
```
Comment (xDivisionByZerox) [inline test/internal/mersenne.spec.ts:131]: Can we not simply do `toBeLessThan(1)`? Looking at the description that would be more clear for me than having a "random" float here.

File: test/modules/date.spec.ts [hunk 4]
```diff
@@ -801,22 +801,26 @@ describe('date', () => {
       faker.setDefaultRefDate(() => new Date(Date.UTC(2020, 0, 1)));
       faker.seed(20200101);
       const date = faker.date.past();
-      expect(date).toEqual(new Date('2019-02-25T21:52:41.824Z'));
+      expect(date).toBeInstanceOf(Date);
+      expect(date).toMatchInlineSnapshot('2019-02-25T21:52:41.824Z');
```
Comment (xDivisionByZerox) [inline test/modules/date.spec.ts]: Why did you change these tests? They seem unrelated to the actual change.

File: test/modules/helpers.spec.ts [hunk 5]
```diff
@@ -1253,6 +1245,24 @@ Try adjusting maxTime or maxRetries parameters for faker.helpers.unique().`)
     }
   );
 
+  describe('uniqueArray()', () => {
```
Comment (xDivisionByZerox) [inline test/modules/helpers.spec.ts]: Is this change required for the test addition?

File: test/internal/mersenne.spec.ts (hunk 2 above)
Comment (ST-DDT) [reply test/internal/mersenne.spec.ts]: They test that we can generate values really close to `1` instead of just `0.9`.

Comment (xDivisionByZerox) [reply test/internal/mersenne.spec.ts]: But you mock the implementation. So you test that the mock returns something what you just said it should return. 🤔

Comment (ST-DDT) [reply test/internal/mersenne.spec.ts]: We test for that value 16 lines above.

Comment (ST-DDT) [reply test/internal/mersenne.spec.ts]: I added a comment to the mock why it is there.

File: test/internal/mersenne.spec.ts (hunk 3 above)
Comment (ST-DDT) [reply test/internal/mersenne.spec.ts:131]: Added the secondary verification as well.

File: test/modules/date.spec.ts (hunk 4 above)
Comment (ST-DDT) [reply test/modules/date.spec.ts]: Reverted, stashed for a later PR.

File: test/modules/helpers.spec.ts (hunk 5 above)
Comment (ST-DDT) [reply test/modules/helpers.spec.ts]: Reverted and stashed for a later PR.

File: test/faker.spec.ts (hunk 1 above)
Comment (ST-DDT) [reply test/faker.spec.ts]: Reverted and stashed for a later PR.

**Review by xDivisionByZerox — APPROVED** [review]

**Review by ejcheng — APPROVED** [review]
